src/Microsoft.Health.Dicom.SqlServer/Registration/DicomServerBuilderSqlServerRegistrationExtensions.cs

```diff
@@ -17,34 +17,53 @@
 using Microsoft.Health.Dicom.SqlServer.Features.Schema;
 using Microsoft.Health.Dicom.SqlServer.Features.Store;
 using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.SqlServer;
 using Microsoft.Health.SqlServer.Api.Registration;
 using Microsoft.Health.SqlServer.Configs;
+using Microsoft.Health.SqlServer.Features.Client;
 using Microsoft.Health.SqlServer.Features.Schema;
+using Microsoft.Health.SqlServer.Features.Schema.Manager;
+using Microsoft.Health.SqlServer.Features.Storage;
 using Microsoft.Health.SqlServer.Registration;
 
 namespace Microsoft.Extensions.DependencyInjection
 {
     public static class DicomServerBuilderSqlServerRegistrationExtensions
     {
         public static IDicomServerBuilder AddSqlServer(
-            this IDicomServerBuilder dicomServerBuilder,
-            IConfiguration configurationRoot,
-            Action<SqlServerDataStoreConfiguration> configureAction = null)
+           this IDicomServerBuilder builder,
+           IConfiguration configurationRoot,
+           Action<SqlServerDataStoreConfiguration> configureAction = null,
+           bool initializeSchema = true)
         {
-            EnsureArg.IsNotNull(dicomServerBuilder, nameof(dicomServerBuilder));
-            IServiceCollection services = dicomServerBuilder.Services;
-
-            services.AddSqlServerBase<SchemaVersion>(configurationRoot);
-            services.AddSqlServerApi();
+            EnsureArg.IsNotNull(builder, nameof(builder));
+            IServiceCollection services = builder.Services;
+            services.AddSqlServerCommon(configurationRoot, configureAction);
+            if (initializeSchema)
```

Comment:
This name seems off. It's deciding way more than whether to initialize the schema or not.

Reply:
I wonder if some sort of broken down composition makes more sense.

Reply:
The problem comes from Azure Function doesn't support `IHostedService` (https://github.com/Azure/azure-functions-host/issues/5447#issuecomment-575368316) -- `SchemaInitializer` is an hosted service. Because of this, we can not use `AddSqlServerBase`.

One thought is Azure Function don't need to upgrade Database schema, the ability to connect is good enough. so the solution would be only initialize those services/components required for connection. The best place the made the change should be in shared component, but as this moment, the feature has not completed yet, these code could change dramatically. Thus I duplicated code from  https://github.com/microsoft/healthcare-shared-components/blob/af69cf5660398ffe2980dd31cca6443e1ee7d433/src/Microsoft.Health.SqlServer/Registration/SqlServerBaseRegistrationExtensions.cs#L22 as a start point.

I don't have a good solution for this, one possible solution could be: separate `AddSqlServerBase` into 2 parts -- utilities for SqlConnection (`PollyRetryLoggerFactory`, `RetrySqlCommandWrapperFactory` etc) and schemaUpgrade (`SchemaInitializer`, `SchemaJobWorker` etc.)